src/createManifest.js

```
@@ -11,163 +10,178 @@
       const resourceEntry = {};
-      resourceEntry.path = resourcesArr[i];
-      // timestamp is optional value, only add if last-modified available
-      const date = resp && resp.headers.get('last-modified');
-      if (ManifestGenerator.isMedia(resourceSubPath)) {
+      resourceEntry.path = resourcePath.trim();
+
+      // Media resources have hash and do not need a timestamp to track changes
+      if (PathUtils.isMedia(resourceEntry.path)) {
+        resourceEntry.hash = PathUtils.getHashFromMedia(resourceEntry.path);
         resourceEntry.path = parentPath.concat(resourceEntry.path);
-        resourceEntry.hash = ManifestGenerator.getHashFromMedia(resourceSubPath);
-      } else if (date) {
-        const timestamp = new Date(date).getTime();
-        if (timestamp > lastModified) {
-          lastModified = timestamp;
+      } else {
+        try {
+          resourceEntry.timestamp = await this.getLastModified(resourceEntry.path);
+          lastModified = Math.max(lastModified, resourceEntry.timestamp);
+        } catch (e) {
+          console.log(`resource ${resourceEntry.path} not available for channel ${channelPath}`);
+          // skip this resource - do not add it as a manifest entry
+          return;
         }
-        resourceEntry.timestamp = timestamp;
       }
       entriesJson.push(resourceEntry);
-    }
+    }));
 
     return [entriesJson, lastModified];
   };
 
-  static createManifest = async (host, manifestMap, path, isHtmlUpdatedMap, additionalAssets = []) => {
-    const data = manifestMap.get(path);
-    /* eslint-disable object-curly-newline */
-    const {
+  /**
+   *
+   * @param {string} channelPath - The path of the channel
+   * @param {[string]} additionalAssets - Additional resources added by generator
+   * @returns Manifest for the channel
+   */
+  createManifestForChannel = async (channelPath, additionalAssets = []) => {
+    // unwrap indexed manifest
```

Lets add a null check here if `channelPath` exists in `indexedManifestsMap` to avoid issues like fragment link is not valid

---

good catch. I think you faced this once when ordering of publish for fragment and parent channel resulted in an error.

Tried to silently ignore incorrect fragment links with this change. other changes(if required) to handle this issue should go into another pr to limit the scope of changes here.